api/app/auth.py

```diff
@@ -66,3 +66,14 @@ async def auth_with_set_fire_starts_permission_required(token=Depends(authentica
             headers={'WWW-Authenticate': 'Bearer'}
         )
     return token
+
+
+async def auth_with_select_station_role_required(token=Depends(authentication_required)):
```

Comment:
The only difference this function has is checking for hfi_select_station instead of hfi_set_fire_starts, can we re-factor into a common function? You'd still retain these two functions, but they could be one liners that call the common function with the roles to check for.